include/iarc7_fc_comms/CommonFcComms.hpp

```diff
@@ -236,7 +248,32 @@ FcCommsReturns CommonFcComms<T>::init()
         return FcCommsReturns::kReturnError;
     }
 
+    contact_switch_subscriber = nh_.subscribe("landing_gear_contact_switches",
+                                         100,
+                                         &CommonFcComms::contactSwitchMessageHandler,
+                                         this);
+
     ROS_INFO("FC Comms registered and subscribed to topics");
+
+    const ros::Time start_time = ros::Time::now();
+    while (ros::ok()
+           && last_contact_switch_message_ptr_ == nullptr
+           && ros::Time::now()
+              < start_time
+                + ros::Duration(CommonConf::kContactSwitchStartupTimeout)) {
+        ros::spinOnce();
+        ros::Duration(0.005).sleep();
+    }
+
+    if (last_contact_switch_message_ptr_ == nullptr)
+    {
+        ROS_WARN("Contact switch message not received within the startup timeout");
```

Comment:
That's definitely more engineering work than this is worth. It requires updating everything that uses fc comms because its an interface change.

The velocity publisher will yell if the foot switches aren't working.

It's not necessary for fc comms operation to have the foot switches.